From 282140b9d77f1e1b0638a8d6454f7062720513f3 Mon Sep 17 00:00:00 2001
From: Demi Marie Obenour <demi@invisiblethingslab.com>
Date: Tue, 23 Mar 2021 12:24:56 -0400
Subject: [PATCH 09/22] Reject RPMv3 packages
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

RPMv3 packages do not have an immutable region trailer, while RPMv4
packgaes do.  Modern rpmsign isn’t even able to sign RPMv3 packages, so
we don’t need to worry about needing to support them in dom0.
---
 lib/header.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/header.c b/lib/header.c
index b16e4e0..569d4aa 100644
--- a/lib/header.c
+++ b/lib/header.c
@@ -1831,7 +1831,9 @@ static rpmRC hdrblobVerifyRegion(rpmTagVal regionTag, int exact_size,
 
     /* Is there an immutable header region tag? */
     if (!(einfo.tag == regionTag)) {
-	rc = RPMRC_NOTFOUND;
+	/* Reject RPMv3 packages */
+	if (!exact_size)
+	    rc = RPMRC_NOTFOUND;
 	goto exit;
     }
 
-- 
2.30.2

